tests/light/src/common/file.py

```diff
@@ -101,8 +104,18 @@ def write(self, content):
         self.__opened_file.flush()
 
     def wait_for_lines(self, lines, timeout=DEFAULT_TIMEOUT):
+        def read_whole_line(f, previous_data=None):
+            if previous_data:
+                buffer = previous_data
+            else:
+                buffer = ""
+            buffer += f.readline()
+            if buffer and buffer[-1] != "\n":
+                return read_whole_line(f, previous_data=buffer)
```

Comment:
Thanks for the notice, it is true. But for now I would keep this as it is.

Reply:
1000 is very small in my opinion.

Recursion can be transformed into a loop easily:
```
def read_whole_line(f):
    buffer = ""
    while buffer[-1] != "\n":
        buffer += f.readline()

    return buffer
```

But I'm not sure we need `read_whole_line()` at all.
`f.readline()` is guaranteed to return a full line unless it's the end of the file.

Reply:
In my opinion this it is a very rare case when this recursion called ~1000 times. 
The recursion happens not for each line, but only when a single line does not contains a new-line character.
It means the write is really slow even a single line could not written, which is weird and not a usual thing. I think if it happens we should investigate this as a separate "case".

Reply:
As I mentioned, `f.readline()` read a full line, I think there is no such case, unless it is the end of the file.

Reply:
> I think if it happens we should investigate it as a separate "case".

I don't want to open the scope of your pull request, we just need to agree on whether this code is invalid or not, because it has been introduced in this pull request.

My understanding is that this whole loop/recursion is unnecessary because it is documented in the Python docs that `f.readline()` always returns with a full line. The only exception is when we reach the end of the file.

Reply:
You mean this can be a good change?
```python
         def find_lines_in_file(lines_to_find, lines_found, f):
-            line_read = read_whole_line(f)
+            line_read = f.readline()
             if not line_read and lines_to_find:
```

Reply:
Sounds good to me. :)